core/flyout_base.js

```diff
@@ -847,3 +847,24 @@ Blockly.Flyout.prototype.placeNewBlock_ = function(oldBlock) {
   block.moveBy(finalOffset.x, finalOffset.y);
   return block;
 };
+
+/**
+ * Handles the given action.
+ * This is only triggered when keyboard accessibility mode is enabled.
+ * @param {!Blockly.Action} action The action to be handled.
+ * @return {boolean} True if the flyout handled the action, false otherwise.
+ * @package
+ */
+Blockly.Flyout.prototype.onBlocklyAction = function(action) {
+  var cursor = this.workspace_.getCursor();
```

Comment:
As discussed, this should just pass it through to the cursor and customize the behavior in the cursor instead.